ui/components/Navigator.js

```diff
@@ -1116,300 +1119,314 @@ class Navigator extends React.Component {
         <OpenInNewIcon style={{ width: '0.85rem', verticalAlign: 'middle' }} />
       </a>
     );
-  }
-
-  render() {
-    const { classes, isDrawerCollapsed, ...other } = this.props;
-    const { path, showHelperButton, navigatorComponents } = this.state;
-    this.updatenavigatorComponentsMenus();
+  };
 
-    const Title = (
-      <div
-        style={
-          !this.state.capabilitiesRegistryObj?.isNavigatorComponentEnabled([DASHBOARD])
-            ? cursorNotAllowed
-            : {}
-        }
-      >
-        <ListItem
-          component="a"
-          onClick={this.handleTitleClick}
-          className={classNames(
-            classes.firebase,
-            classes.item,
-            classes.itemCategory,
-            classes.cursorPointer,
-            !this.state.capabilitiesRegistryObj?.isNavigatorComponentEnabled([DASHBOARD]) &&
-              classes.disableLogo,
-          )}
-        >
-          <img
-            className={isDrawerCollapsed ? classes.mainLogoCollapsed : classes.mainLogo}
-            src="/static/img/meshery-logo.png"
-            onClick={this.handleTitleClick}
-          />
-          <img
-            className={isDrawerCollapsed ? classes.mainLogoTextCollapsed : classes.mainLogoText}
-            src="/static/img/meshery-logo-text.png"
-            onClick={this.handleTitleClick}
-          />
+  useEffect(() => {
+    updatenavigatorComponentsMenus();
+  }, [updatenavigatorComponentsMenus]);
```

Comment:
It should be, if not already
```suggestion
  }, [updateNavigatorComponentsMenus]);
```

Reply:
it was already like this. should I refactor it?

Reply:
Yes please, if this is only in this file, yes